libs/gi/page-team/src/CharacterDisplay/Tabs/TabTheorycraft/ArtifactSubCard/ArtifactAllSubstatEditor.tsx

```diff
@@ -120,28 +120,42 @@ export function ArtifactAllSubstatEditor({
           disabled={disabled}
         />
       </CardThemed>
-      <CustomNumberInput
-        color={rolls && invalid ? 'warning' : 'primary'}
-        float
-        startAdornment={
-          <Box sx={{ whiteSpace: 'nowrap' }}>{t`tabTheorycraft.all.rolls`}</Box>
-        }
+      <NumberInputLazy
         value={parseFloat((rolls ?? 0).toFixed(2))}
+        float
         onChange={(v) => v !== undefined && setRolls([v])}
-        sx={{ borderRadius: 1, px: 1, my: 0, height: '100%', width: '7em' }}
-        inputProps={{ sx: { textAlign: 'right', pr: 0.5 }, min: 0, step: 1 }}
-        disabled={disabled}
+        color={rolls && invalid ? 'warning' : 'primary'}
+        size="small"
+        inputProps={{
+          sx: { textAlign: 'right', width: '1.5em' },
+          min: 0,
+          max: 99,
```

Comment:
Probably should add validation on the datamanager side. but since we didn't validate <0 values afaik, it's fine to continue to not validate for now